src/core/text-buffer.cc

```diff
@@ -671,18 +676,17 @@ struct TextBuffer::Layer {
 };
 
 TextBuffer::TextBuffer(u16string &&text) :
-  base_layer{new Layer(move(text))},
+  base_layer{make_shared<Layer>(Layer(move(text)))},
```

Comment:
Not sure if calling Layer() explicitly is required (instead of just writing make_shared).